pymtl3/passes/PrintWavePass.py

```diff
@@ -0,0 +1,127 @@
+"""
+========================================================================
+PrintWavePass.py
+========================================================================
+Print the most sigficant number of signals in wave form.
+top object uses top._print_wave(top) to print.
+
+Author : Kaishuo Cheng
+Date   : Oct 4, 2019
+"""
+
+from __future__ import absolute_import, division, print_function
+import time
+from collections import defaultdict
+from copy import deepcopy
+import py
+import sys
+from pymtl3.dsl import Const
+from pymtl3.passes.BasePass import BasePass, PassMetadata
+import random
+
+class PrintWavePass( BasePass ):
+
+  def __call__( self, top ):
+    setattr(top,"_print_wave",_help_print)
+
+def _process_binary(sig,base):
+    """
+    Returns int value from a signal in 32b form. Used for testing.
+
+    Example: input: 0b00000000000000000000000000000000 10
+             output: 0
+    """
+    if sig[1] == "b":
+        sig = sig[2:]
+    if base == 10:
+        tempint = int(sig,2)
+        if sig[0] == '1':
+            return tempint -2 **32      #taking 2's complement.
+                                    #leading 1 indicates a negative number
+        else:
+            return tempint
+    #hex number
+    else:
+        temphex = hex(int(sig,2))
+        l = len(temphex)
+        if l > 4:
+            temphex = temphex[:4]
+        if l < 4:
+            temphex = '0'*(4-l) + temphex
+        return temphex
+
+def _help_print(self):
+    char_length = 5
+    _tick = u'\u258f'
+    _up, _down = u'\u2571', u'\u2572'
+    _x, _low, _high = u'\u2573', u'\u005f', u'\u203e'
+    _revstart, _revstop = '\x1B[7m', '\x1B[0m'
+    _lightgrey = '\033[47m'
+    _back='\033[0m'  #back to normal printing
+    allsignals = self._collectsignals
+
+    #spaces before cycle number
+    maxlength = 0
+    for sig in allsignals:
+        #   Example: s.in(12b)
+        # length of signal name + (b) + number of digits, like 12
+        thislength = len(sig) + len(str(len(allsignals[sig][0][0])))+3
+        if thislength > maxlength:
+            if sig != "s.clk" and sig != "s.reset":
+                maxlength = thislength
+
+    print(" "*(maxlength+1),end = "")
+
+    for i in range(0,len(allsignals["s.clk"]),5):
+          print(_tick + str(i)+ " "*(5*char_length-1) ,end="")
+
+    #signals
+    for sig in allsignals:
```

Comment:
we usually put underscore between words. For example I would suggest ```all_signals``` here

Reply:
I guess you can take a pass through the whole PR and improve the variable names

Reply:
Got it, I will work on that